src/components/ImportEmailsDialog.vue

```diff
@@ -0,0 +1,211 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+
+import IconFileUpload from 'vue-material-design-icons/FileUpload.vue'
+
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import { t, n } from '@nextcloud/l10n'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcDialog from '@nextcloud/vue/dist/Components/NcDialog.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+import { importEmails } from '../services/participantsService.js'
+
+const loading = ref(false)
+const listImport = ref(null)
+
+const props = defineProps<{
+	token: string,
+}>()
+const emit = defineEmits<{
+	(event: 'close'): void,
+}>()
+
+const importedFile = ref<File>(null)
+const uploadResult = ref(null)
+const uploadResultCaption = computed(() => {
+	return uploadResult.value?.error
+		? { class: 'import-list__caption--error', label: t('spreed', 'Error while verifying uploaded file') }
+		: { class: 'import-list__caption--success', label: t('spreed', 'Uploaded file is verified') }
+})
+
+/**
+ * Call native input[type='file'] to import a file
+ */
+function triggerImport() {
+	listImport.value.value = ''
+	listImport.value.click()
+}
+
+/**
+ * Validate imported file and insert data into form fields
+ * @param event import event
+ */
+function importList(event: Event) {
+	const file = (event.target as HTMLInputElement).files?.[0]
+	if (!file) {
+		importedFile.value = null
+	}
+
+	importedFile.value = file
+
+	testList(importedFile.value)
+}
+
+/**
+ * Verify imported file and show results
+ * @param file file to upload
+ */
+async function testList(file: File) {
+	loading.value = true
+	uploadResult.value = null
+	try {
+		const response = await importEmails(props.token, file, true)
+		uploadResult.value = response.data.ocs.data
+	} catch (error) {
+		uploadResult.value = error?.response?.data?.ocs?.data
+	} finally {
+		loading.value = false
+	}
+}
+
+/**
+ * Verify imported file and add participants
+ * @param file file to upload
+ */
+async function submitList(file: File) {
+	if (!file) {
+		return
+	}
+
+	try {
+		await importEmails(props.token, file, false)
+		showSuccess(t('spreed', 'Participants added successfully'))
```

Comment:
this should also close the dialog
![grafik](https://github.com/user-attachments/assets/39e81908-21a4-4fdf-9a79-0a80ae8f223c)

Reply:
fixed